cmd/experimental/kjobctl/pkg/parser/time_limit_test.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package parser
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"k8s.io/utils/ptr"
+)
+
+func TestParseTimeLimit(t *testing.T) {
+	testCases := map[string]struct {
+		val     string
+		want    *int32
+		wantErr error
+	}{
+		"empty value": {},
+		"-1": {
+			val: "-1",
+		},
+		"INFINITE": {
+			val: "INFINITE",
+		},
+		"UNLIMITED": {
+			val: "UNLIMITED",
+		},
+		"infinite": {
+			val: "infinite",
+		},
+		"unlimited": {
+			val: "unlimited",
+		},
```

Comment:
please, excuse my ignorance, I'm probably missing something but don't we need `want`s here?